padertorch/ops/losses/loss.py

```diff
@@ -105,12 +131,19 @@ def pit_loss(estimate, target, loss_fn=torch.nn.functional.mse_loss):
             f'{estimate.size()} != {target.size()}'
         )
     candidates = []
+    filler = (slice(None),) * axis 
     for permutation in itertools.permutations(range(sources)):
-        candidates.append(loss_fn(
-            estimate[:, permutation, :],
+        candidates.append((loss_fn(
+            estimate[filler + (permutation, )],
             target
-        ))
-    return torch.min(torch.stack(candidates))
+        ), permutation))
+
+    min_loss, min_perm = min(candidates, key=lambda x: x[0])
```

Comment:
`torch.min` has the advantage that the execution is async on the GPU.
Python `min` introduces a sync point. It would be nice to have no sync point.

Reply:
It is not possible to have no sync point if we want to output the permutation (right?). We could use `torch.argmax` and then index into the list of permutations in case the permutation is requested. Then we would at least skip the sync point when the permutation is not requested

Reply:
You are right. When the permutation is needed, we have to sync the GPU and the CPU.
Since in many cases we do not need the permutation, `torch.argmax` would be nicer, because then pytorch has the chance to do the operation without a sync.